src/client/powerpages/utils/MultiStepInput.ts

```diff
@@ -0,0 +1,190 @@
+    static async run(start: InputStep) {
+        const input = new MultiStepInput();
+        return input.stepThrough(start);
+    }
+
+    private current?: QuickInput;
+    private steps: InputStep[] = [];
+
+    private async stepThrough(start: InputStep) {
+        let step: InputStep | void = start;
+        while (step) {
+            this.steps.push(step);
+            if (this.current) {
+                this.current.enabled = false;
+                this.current.busy = true;
+            }
+            try {
+                step = await step(this);
+            } catch (err) {
+                if (err === InputFlowAction.back) {
+                    this.steps.pop();
+                    step = this.steps.pop();
+                } else if (err === InputFlowAction.resume) {
+                    step = this.steps.pop();
+                } else if (err === InputFlowAction.cancel) {
+                    step = undefined;
+                } else {
+                    throw err;
+                }
+            }
+        }
+        if (this.current) {
+            this.current.dispose();
+        }
+    }
+
+    async showQuickPick<
+        T extends QuickPickItem,
+        P extends QuickPickParameters<T>
+    >({ title, step, totalSteps, items, activeItem, placeholder, buttons }: P) {
+        const disposables: Disposable[] = [];
+        try {
+            return await new Promise<
+                T | (P extends { buttons: (infer I)[] } ? I : never)
+            >((resolve, reject) => {
+                const input = window.createQuickPick<T>();
+                input.title = title;
+                input.step = step;
+                input.totalSteps = totalSteps;
+                input.placeholder = placeholder;
+                input.items = items;
+                if (activeItem) {
+                    input.activeItems = [activeItem];
+                }
+                input.buttons = [
+                    ...(this.steps.length > 1 ? [QuickInputButtons.Back] : []),
+                    ...(buttons || []),
+                ];
+                disposables.push(
+                    input.onDidTriggerButton((item) => {
+                        if (item === QuickInputButtons.Back) {
+                            reject(InputFlowAction.back);
+                        } else {
+                            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                            resolve(<any>item);
+                        }
+                    }),
+                    input.onDidChangeSelection((items) => resolve(items[0]))
+                );
+                if (this.current) {
+                    this.current.dispose();
+                }
+                this.current = input;
+                this.current.show();
+            });
+        } finally {
+            disposables.forEach((d) => d.dispose());
+        }
+    }
+
+    async showInputBox<P extends InputBoxParameters>({
+        title,
```

Comment:
Naming convention looks very naive in most of the method

Reply:
I have updated some custom method names but for vscode related methods I tried sticking to their docs and example implementations. Let me know if some specific method name should be changed.